src/frontend/src/session/cursor_manager.rs

```diff
@@ -740,6 +742,52 @@ impl SubscriptionCursor {
         Ok(row)
     }
 
+    pub fn process_output_desc_row(
```

Comment:
Given that the `Field` and the row structure (i.e. which columns in which idx) are fixed, there are several things can be simplified in this method:
- We can use `row.project(<pk_col_idx_in_row>)` to get the pk instead of doing the filter_map and map checking every time.
- We only need to calculate the `pk_fields` once and cache it in SubscriptionCursor instead of doing it every time when we get rows.